### DISCUSSION TITLE
Disable validation of group of vields depend on other field OR external variable

### DISCUSSION THREAD

**Discussion by inomn** [react-hook-form/react-hook-form#9468] [Q&A]
Hi! I have a relatively complex form, and some parts of it depend on a user's choices. As a simple example, basic checkout form usually contains the radio selector "is billing address the same as delivery address", and if "no", you get an additional set of fields, and now they should be validated. But if "yes", then form should be valid even if those fields are empty \ invalid.
How to implement it with react-hook-form? Ideally for me, I want to refer not to another field in form, but to an external variable to check should I validate part of the form or not.

Thanks.

**Comment by bluebill1049** [comment]
If you are using native inputs, i think you can use `<fieldset />` and set it to `disabled`.

**Reply by inomn** [reply]
Thanks for your reply. I'm using `<Controller>` with custom input in it. And looks like `shouldUnregister` doesn't work. For still does not possible to submit.  Form state is not valid, and in errors object all those "disabled and shouldUnregistered" fields. Any ideas of how to solve my issue?
